process/block/shardblock.go

```diff
@@ -921,6 +923,38 @@ func (sp *shardProcessor) updateState(headers []data.HeaderHandler) {
 	}
 }
 
+func (sp *shardProcessor) snapShotEpochStartFromMeta(header *block.Header) {
+	accounts := sp.accountsDB[state.UserAccountsState]
+	if !accounts.IsPruningEnabled() {
+		return
+	}
+
+	if header.IsStartOfEpochBlock() {
+		epochStartId := core.EpochStartIdentifier(header.GetEpoch())
+		metaBlock, err := process.GetMetaHeaderFromStorage([]byte(epochStartId), sp.marshalizer, sp.store)
+		if err != nil {
+			log.Warn("could not find epoch start metablock for", "epoch", header.GetEpoch(), "err", err)
+			return
+		}
+
+		for _, epochStartShData := range metaBlock.EpochStart.LastFinalizedHeaders {
+			if epochStartShData.ShardId != header.ShardId {
+				continue
+			}
+
+			rootHash := epochStartShData.RootHash
+			accounts.CancelPrune(rootHash, data.NewRoot)
+			log.Debug("shard trie snapshot from epoch start shard data", "rootHash", rootHash)
+			accounts.SnapshotState(rootHash)
+
+			return
+		}
+
+		log.Warn("could not find epoch start shard data in metaBlock for", "epoch", header.GetEpoch(), "err", err)
```

Comment:
log.Debug

Reply:
I would leave as log.Warn. it should never happen, only if there was something incorrectly coded.

Reply:
ok